pkg/ddevapp/global_dotddev_assets/commands/host/launch

```diff
@@ -52,15 +52,16 @@ while :; do
      shift
  done
 
-if [ -n "${1:-}" ] ; then
-  if [[ ${1::1} != "/" ]] ; then
-    FULLURL="${FULLURL}/";
-  fi
-
+if [ -n "${1:-}" ]; then
   if [[ $1 =~ ^https?:// ]]; then
+    # full url
     FULLURL="${1}";
+  elif [[ $1 =~ ^: ]]; then
+    # specific port
+    FULLURL="${FULLURL%:[0-9]*}${1}";
```

Comment:
What happens if the port is for `http` but the `FULLURL` is https? Or the other way around?

Reply:
The protocol is not changed on ddev (might be redirected by e.g. nginx), so you can end on

```
400 Bad Request
The plain HTTP request was sent to HTTPS port
nginx
```

Reply:
That's what I thought was probably the case.
It feels like this PR is missing a trick there. With the original PR (and the [alternative I suggested](https://github.com/ddev/ddev/pull/5525#issuecomment-1979759118)) checking HTTP vs HTTPS and setting the correct port is managed, so we don't get into this situation.

Right now `ddev launch` has logic to check whether it should be using HTTP or HTTPS - but that logic isn't going to be leveraged here.

Reply:
Looking at [ddev adminer](https://github.com/ddev/ddev-adminer/blob/main/commands/host/adminer) for example, there's just as much code figuring out what the URL should be (http vs https etc) as there is launching it.

Reply:
> What happens if the port is for http but the FULLURL is https? Or the other way around?
> It feels like this PR is missing a trick there. 

Good point, thank you. I added detection for http/https.

Reply:
> Looking at [ddev adminer](https://github.com/ddev/ddev-adminer/blob/main/commands/host/adminer) for example, there's just as much code figuring out what the URL should be (http vs https etc) as there is launching it. Ideally that could all be abstracted away together.

The port for adminer can be shortened right now to (assuming someone is using DDEV v1.23.0-alpha1):

```bash
DDEV_ADMINER_PORT=9100
DDEV_ADMINER_HTTPS_PORT=9101

if [ ${DDEV_PRIMARY_URL%://*} = "https" ]; then
    ddev launch $DDEV_PRIMARY_URL:$DDEV_ADMINER_HTTPS_PORT
else
    ddev launch $DDEV_PRIMARY_URL::$DDEV_ADMINER_PORT
fi
```

or after this PR:

```bash
DDEV_ADMINER_PORT=9100
DDEV_ADMINER_HTTPS_PORT=9101

if [ ${DDEV_PRIMARY_URL%://*} = "https" ]; then
    ddev launch :$DDEV_ADMINER_HTTPS_PORT
else
    ddev launch :$DDEV_ADMINER_PORT
fi
```

But I'm not going to change it in the near future, because people can use the old DDEV with the new add-on version. And we have no restrictions on this. Ideally, add-ons should work like composer packages and pull only compatible add-on version, but at the moment that's probably too much work.